### PR TITLE
cluster info config map

### PR DESCRIPTION
**Pull request by stevenctl**
Proof of concept, needs cleanup. 

- On install (base component), create a configmap with cluster name and network from install values. 
- Setting up a remote cluster needs `base` installed before the remote secret is created. We might be able to deal with this by setting up a watch for the config map. 
- When creating a kube registry, read the config map to allow overriding cluster name and optionally overriding network name in a way that does not require updating values in every connected cluster. 

We might want to use this in injection as well for setting `ISTIO_META_CLUSTER_*` when that's not set via url values. 

This fixes tests that reach `naked` service in  [integ-pilot-multicluster-tests_istio](https://prow.istio.io/?repo=istio%2Fistio&pull=28035&job=integ-pilot-multicluster-tests_istio).

### CODE REVIEW

File: manifests/charts/istio-control/istio-discovery/templates/configmap.yaml [hunk 1]
```diff
@@ -88,3 +88,20 @@ data:
...
+{{- if .Values.global.multiCluster.clusterName }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: istio-cluster{{- if not (eq .Values.revision "") }}-{{ .Values.revision }}{{- end }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    istio.io/rev: {{ .Values.revision | default "default" }}
+    release: {{ .Release.Name }}
+    istio/multiCluster: true
```
Comment (stevenctl) [inline manifests/charts/istio-control/istio-discovery/templates/configmap.yaml:101]: Why doesn't this label get applied? The above labels along with some other things that look like they came from operator seem to be there but this gets removed.

File: manifests/charts/base/templates/configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+metadata:
+  name: istio-cluster{{- if .Values.revision }}-{{ .Values.revision }}{{- end }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    istio.io/rev: {{ .Values.revision | default "default" }}
+    release: {{ .Release.Name }}
+    istio/multiCluster: true
+data:
+  cluster: {{ .Values.global.multiCluster.clusterName }}
+{{- if .Values.global.network }}
```
Comment (nmittler) [inline manifests/charts/base/templates/configmap.yaml:13]: do you even need the if?

**Comment by stevenctl** [comment]
closing in favor of #28081

**Pull request closed without merging**